src/Chapter3.hs

```diff
@@ -480,6 +519,37 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+
+data Building = Church | Library deriving (Show)
+data House = One | Two | Three | Four deriving (Enum, Show)
+
+data City = City
+    { cityCastle :: Maybe Castle
+    , cityWalls  :: Bool
+    , cityBuilding  :: Building
+    , cityHouses  :: [House]
+    } deriving Show
+
+newtype Castle = Castle String deriving Show
+
+buildCastle :: String -> City -> City
+buildCastle castleName city =
+    city { cityCastle = Just (Castle castleName) }
+
+buildHouse :: City -> City
+buildHouse city =
+    city { cityHouses = Two : cityHouses city }
```

Comment:
Nice one! 🏠 

If you want to be able to build any type of the house, you can just pass it into the function itself. So the type could also be:

```haskell
buildHouse :: House -> City -> City
```